pkg/controller/utils/component.go

```diff
@@ -446,9 +446,10 @@ func mergeState(desired client.Object, current runtime.Object) client.Object {
 		cj := current.(*batchv1.Job)
 		dj := desired.(*batchv1.Job)
 
-		// We're only comparing jobs based off of annotations for now so we can send a signal to recreate a job. Later
-		// we might want to have some better comparison of jobs so that a changed in the container spec would trigger
-		// a recreation of the job
+		if cj.Spec.Template.Spec.Containers[0].Image != dj.Spec.Template.Spec.Containers[0].Image {
```

Comment:
@rene-dekker Is it a safe assumption here that there will always be one image in the spec?

Reply:
I think you should check for length and iterate over the containers to avoid a potential bug in the future. Otherwise, I think this fix will suffice.